common/mqtt_client/tests/packet_size_tests.rs

```diff
@@ -0,0 +1,178 @@
+    // Start the local broker
+    let mqtt_server_handle = tokio::spawn(async {
+        start_broker_local("../../configuration/rumqttd/rumqttd_1883.conf").await
+    });
+    // Start the subscriber
+    let subscriber = tokio::spawn(async move { subscribe_messages().await });
+
+    // Start the publisher and publish 3 messages
+    let publisher = tokio::spawn(async move { publish_messages().await });
+
+    let _ = publisher.await?;
+    let res = subscriber.await?;
+    mqtt_server_handle.abort();
+    match res {
+        Err(e) => {
+            return Err(e);
+        }
+        _ => {
+            return Ok(());
+        }
+    }
+}
+
+#[tokio::test]
+// This checks the mqtt packet size that exceeds the limit
+async fn packet_size_exceeds_limit() -> Result<(), anyhow::Error> {
+    // Start the broker
+    let mqtt_server_handle = tokio::spawn(async {
+        start_broker_local("../../configuration/rumqttd/rumqttd_1884.conf").await
+    });
+
+    // Start the publisher and publish a message
+    let publish = tokio::spawn(async { publish_big_message().await });
+
+    let res = publish.await?;
+    mqtt_server_handle.abort();
+    match res {
+        Err(e) => {
+            return Err(e);
+        }
+        _ => {
+            return Ok(());
+        }
+    }
+}
+
+async fn subscribe_errors(pub_client: &Client) -> Result<(), MqttClientError> {
+    let mut errors = pub_client.subscribe_errors();
+    while let Some(error) = errors.next().await {
+        match *error {
+            MqttClientError::ConnectionError(rumqttc::ConnectionError::MqttState(
+                StateError::Deserialization(rumqttc::Error::PayloadTooLong),
+            )) => {
+                return Err(mqtt_client::MqttClientError::ConnectionError(
+                    rumqttc::ConnectionError::Mqtt4Bytes(rumqttc::Error::PayloadTooLong),
+                ));
+            }
+            _ => {
+                return Ok(());
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn start_broker_local(cfile: &str) -> anyhow::Result<()> {
+    let config: Config = confy::load_path(cfile)?;
+    let (mut router, _console, servers, _builder) = async_locallink::construct_broker(config);
+    let router = tokio::task::spawn_blocking(move || -> anyhow::Result<()> { Ok(router.start()?) });
+    servers.await;
+    let _ = router.await;
+    Ok(())
+}
+
+async fn subscribe_messages() -> Result<(), anyhow::Error> {
+    let sub_filter = TopicFilter::new("test/hello")?;
+    let client = Client::connect("subscribe", &mqtt_client::Config::default()).await?;
+    let mut messages = client.subscribe(sub_filter).await?;
+    let mut cnt: i32 = 0;
+    while let Some(_message) = messages.next().await {
+        if cnt >= 3 {
+            break;
+        } else {
+            cnt += 1;
+        }
+    }
+    assert!(cnt >= 3);
+    client.disconnect().await?;
+    Ok(())
+}
+
+async fn publish_messages() -> Result<(), anyhow::Error> {
```

Comment:
As above, name the function more appropriately, eg:
`publish_large_message_3_times`